js/npc/npc.js

```diff
@@ -260,5 +304,15 @@ setup.getNpcHappyLevel = function(person) {
     }
 
     return 'very_happy';
-}
+};
 
+setup.personalityTraits = function(count = 1) {
+	var c = shuffle(['efficient','organized','extravagant','careless']);
+	var a = shuffle(['friendly','compassionate','critical','rational']);
+	var n = shuffle(['sensitive','nervous','resilient','confident']);
+	var o = shuffle(['inventive','curious','consistent','cautious']);
+	var e = shuffle(['outgoing','energetic','solitary','reserved']);
+	var traits = shuffle([c[0], a[0], n[0], o[0], e[0]]);
+
+	return traits.slice(0, count).sort();
```

Comment:
Why didn't just put them all in one array but instead 5 groups?

Reply:
because we don't want several from the same group.